src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -234,11 +271,21 @@ public static InstructionSetSupport ConfigureInstructionSetSupport(string instru
             {
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("aes");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("crc");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("sha1");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("sha2");
-                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lse");
                 optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("dotprod");
+                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("lse");
+                optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("rcpc");
+
+                if (!isReadyToRun)
+                {
+                    // This is checked for nearly every function since it allows simplified addressing
```

Comment:
I would not expect this to be checked for nearly every function. Is it a bug in the JIT that it checks it for nearly function?

The `InstructionSet_Rcpc2` checks in the JIT seem to be done in a wrong spot. They should be done last - once we know we can take advantage of it. Instead, they are done first that looks like a bug.